x/claims/keeper/abci.go

```diff
@@ -83,18 +83,15 @@ func (k Keeper) ClawbackEscrowedTokens(ctx sdk.Context) error {
 // hasn't performed a single tx during the claim window).
 // Once the account is clawbacked, the claims record is deleted from state.
 func (k Keeper) ClawbackEmptyAccounts(ctx sdk.Context, claimsDenom string) {
-	totalClawback := sdk.Coin{Denom: claimsDenom, Amount: sdk.ZeroInt()}
+	totalClawback := sdk.Coins{}
 	logger := k.Logger(ctx)
 
+	accPruned := int64(0)
 	accClawbacked := int64(0)
 
-	var addresses []sdk.AccAddress
-
 	k.IterateClaimsRecords(ctx, func(addr sdk.AccAddress, _ types.ClaimsRecord) (stop bool) {
-		// NOTE: we cannot delete the record while iterating over it
-		defer func() {
-			addresses = append(addresses, addr)
```

Comment:
We should keep this

Reply:
@fedekunze Could you explain why?

Reply:
> 	// CONTRACT: No writes may happen within a domain while an iterator exists over it.
	// Exceptionally allowed for cachekv.Store, safe to write in the modules.

Ref: https://github.com/cosmos/cosmos-sdk/blob/c2fd51b4c5f41efc56c9aec1f44b4ce9e963dfc3/store/types/store.go#L215-L221